### PR TITLE
FW-4119 Add base model class

### PR DESCRIPTION
**Pull request by NolanVH**
This change adds the base model class which can be inherited by other models. 
The base class contains the following fields:

- id - a UUID
- is_trashed - a field to indicate if the model has been deleted
- created_by - a foreign key to a user
- created - date and time which is automatically updated on creation using the auto_now_add option
- last_modified_by - a foreign key to a user
- last_modified - date and time which is automatically updated when the model is modified using the auto_now option

The date fields are automatically handled and updated but the user fields must be updated when the documents are created or modified.

### CODE REVIEW

**Review by sahil97 — APPROVED** [review]

**Review by bronson-c — APPROVED** [review]
LGTM 👍

File: fv_be/app/models/base.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    is_trashed = models.BooleanField(default=False)
+    created_by = models.ForeignKey(
+        User, on_delete=models.SET_NULL, null=True, default=None
+    )
+    created_by_date = models.DateTimeField(auto_now_add=True)
+    last_modified_by = models.ForeignKey(
+        User, on_delete=models.SET_NULL, null=True, default=None
+    )
+    last_modified_date = models.DateTimeField(auto_now=True)
```
Comment (gmcauliffe) [inline fv_be/app/models/base.py:37]: Is there added value in the explicitness of suffix `_by_date`?
Could we simplify these two from `created_by_date` and `last_modified_by_date`
to `created` and `last_modified` ?

File: fv_be/app/models/base.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+# this will be replaced by our custom User class when we create the site models
+from django.contrib.auth.models import User
+from django.db import models
+from rules.contrib.models import RulesModel
+
+
+class BaseModel(RulesModel):
+    """
+    Base model for all FirstVoices Backend models, with standard fields and support for rules-based permissions.
```
Comment (gmcauliffe) [inline fv_be/app/models/base.py:11]: Very helpful level of detail in these comments 👍

File: fv_be/app/models/base.py (hunk 1 above)
Comment (sarahfirstvoices) [reply fv_be/app/models/base.py:37]: I like the idea of the shorter names

Comment (NolanVH) [reply fv_be/app/models/base.py]: I agree that shorting naming could be used here. I have updated the naming in a small commit. Thanks for the suggestion @gmcauliffe.

**Review by caforbes — APPROVED** [review]
Looking good! :)

File: fv_be/app/models/base.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            "add": rules.is_staff,
+            "view": rules.is_authenticated,
+        }
+    See the django-rules docs [https://github.com/dfunckt/django-rules#permissions-in-models] for more details.
+    """
+
+    class Meta:
+        abstract = True
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
```
Comment (plasticviking) [inline fv_be/app/models/base.py:27]: Is there a reason to prefer UUID rather than a bigint here?

**Pull request merged**